### PR TITLE
FAB-16120 Adding IT for MSP inconsistence

### PR DESCRIPTION
**Pull request by DereckLuo**
Signed-off-by: Chongxin Luo <Chongxin.Luo@ibm.com>

#### Type of change

- Test update

#### Description
- Adding integration test to replace behave test in **fabric-test**. 
- Testing correct endorsement behavior when a peer with _member_ identity trying to endorse a chaincode with _OR (Org1.peer, Org2.peer)_ endorsement policy.
- Minor typo change in integration test.

#### Related issues
https://jira.hyperledger.org/browse/FAB-16120

### CODE REVIEW

File: integration/msp/msp_test.go [hunk 1]
```diff
@@ -58,6 +58,10 @@ var _ = Describe("MSP identity test on a network with mutual TLS required", func
 		By("setting TLS ClientAuthRequired to be true for all peers and orderers")
 		network.ClientAuthRequired = true
 
+		By("disabling NodeOY for org2, and setting all peers in org2 with member identity")
```
Comment (mastersingh24) [inline integration/msp/msp_test.go]:
```suggestion
By("disabling NodeOU for org2, and setting all peers in org2 with member identity")
```

**Review by mastersingh24 — DISMISSED** [review]

File: integration/msp/msp_test.go [hunk 2]
```diff
@@ -58,6 +58,10 @@ var _ = Describe("MSP identity test on a network with mutual TLS required", func
 		By("setting TLS ClientAuthRequired to be true for all peers and orderers")
 		network.ClientAuthRequired = true
 
+		By("disabling NodeOU for org2, and setting all peers in org2 with member identity")
```
Comment (jyellick) [inline integration/msp/msp_test.go]: What does "and setting all peers in org2 with member identity" mean?

**Review by jyellick — COMMENTED** [review]
As best as I can tell, this test is verifying that if you set an endorsement policy to require a peer role, for an org which does not have node OUs enabled, that that org won't be able to validly endorse.  But it's not at all clear from the description of the test that this is the scenario being tested.

It actually seems like it's testing two scenarios, one, where the policy is not satisfied because the peer's signer cert does not have the peer node OU, and one where the policy is not satisfied because the peer's MSP does not define the peer node OU.  It would be nice to explicitly call this out in the test, and to name which orgs/peers are testing which scenarios.

File: integration/nwo/configtx_template.go [hunk 3]
```diff
@@ -26,6 +27,20 @@ Organizations:{{ range .PeerOrgs }}
...
+      Rule: OR('{{.MSPID}}.member')
+    Writers:
+      Type: Signature
+      Rule: OR('{{.MSPID}}.member')
+    Endorsement:
+      Type: Signature
+      Rule: OR('{{.MSPID}}.member')
+    Admins:
+      Type: Signature
+      Rule: OR('{{.MSPID}}.member')
```
Comment (jyellick) [inline integration/nwo/configtx_template.go]: I don't think we want this.  Even without node-OUs, your admins should still require the admin role.

File: integration/nwo/configtx_template.go [hunk 4]
```diff
@@ -50,13 +66,38 @@ Organizations:{{ range .PeerOrgs }}
...
```
Comment (jyellick) [inline integration/nwo/configtx_template.go]: And same here

File: integration/nwo/configtx_template.go [hunk 5]
```diff
@@ -50,13 +66,38 @@ Organizations:{{ range .PeerOrgs }}
...
+    {{- end }}
 {{ end }}
 {{- range .OrdererOrgs }}
 - &{{ .MSPID }}
   Name: {{ .Name }}
   ID: {{ .MSPID }}
   MSPDir: {{ $w.OrdererOrgMSPDir . }}
   Policies:
+  {{- if .EnableNodeOUs }}
+    Readers:
```
Comment (jyellick) [inline integration/nwo/configtx_template.go:91]: I'd think we'd want to remove the '.peer' role from there, and add the .orderer role to both.

**Review by jyellick — CHANGES_REQUESTED** [review]
The bulk of it looks good, but a few issues with the template

**Review by jyellick — APPROVED** [review]

**Review by jyellick — APPROVED** [review]

**Pull request merged**